adam/experiment/live_demo.py

```diff
@@ -0,0 +1,319 @@
+                with open(info_path, encoding="utf-8") as info_file:
+                    info_yaml = yaml.safe_load(info_file)
+                self.situation_number = int(info_yaml["num_dirs"])
+
+        self.curriculum_path.mkdir(parents=True, exist_ok=True)
+
+    def reset(self, *, initialize_after_reset: bool = True) -> None:
+        """Reset the curriculum directory, optionally reinitialize it."""
+        if self.curriculum_path.is_dir():
+            self.curriculum_path.rmdir()
+
+        if initialize_after_reset:
+            self.initialize_curriculum()
+
+    def preprocess(
+        self, image_path: Path, host: str, stroke_model_path: Path, stroke_python: Path
+    ) -> Path:
+        """Preprocess an image with stroke extraction to configure the curriculum scene."""
+        situation_dir = self.curriculum_path / f"situation_{self.situation_number}"
+        situation_dir.mkdir(parents=True)
+        situation_img_path = situation_dir / "rgb_0.png"
+        segmentation_path = situation_dir / "semantic_0.png"
+        img = cv2.imread(str(image_path))
+        img2 = None
+        try:
+            img2 = cv2.cvtColor(img, cv2.COLOR_RGBA2RGB)
+            print("Image is RGBA")
+        except ValueError:
+            print("Image is RGB")
+            img2 = img
+        finally:
+            cv2.imwrite(str(situation_img_path), img2)
+
+        # Generate a Semantic Segmentation Image
+        with situation_img_path.open("rb") as img_file:
+            im_bytes = img_file.read()
+        im_b64 = base64.b64encode(im_bytes).decode("utf8")
+        headers = {"Content-type": "application/json", "Accept": "text/plain"}
+        payload = json.dumps({"image": im_b64, "other_key": "value"})
+        api = host + "instanceseg"
+        print(f"Processing Image Segmentation from host: {api}")
+        resp = requests.post(api, data=payload, headers=headers)
+
+        data = resp.json()
+
+        print("Generating Segmentation Mask")
+        # 3D array: (N, W, H) where N is number of masks
+        masks = np.asarray(data["masks"], dtype=np.uint8)
+        print(f"Got {len(masks)} masks.")
+
+        # Assert masks don't overlap
+        for i in range(len(masks)):  # pylint: disable=consider-using-enumerate
+            for j in range(i + 1, len(masks)):  # pylint: disable=consider-using-enumerate
+                if np.any(masks[i] & masks[j]):
+                    print(
+                        f"Warning: Masks {i} and {j} overlap. Preferring later mask {j}."
+                    )
+
+        # Compute segmentation IDs
+        segmentation_ids = (
+            np.max(
+                masks
+                * (1 + np.arange(masks.shape[0], dtype=masks.dtype))[
+                    :, np.newaxis, np.newaxis
+                ],
+                axis=0,
+            )
+            if masks.size > 0
+            else np.zeros(img.shape[:2], dtype=np.uint32)
+        )
+        rng = np.random.default_rng(self.color_seed)
```

Comment:
Guide comment on what we're doing with the IDs?

```suggestion
        # Create segmentation mask image by assigning random colors to each mask
        rng = np.random.default_rng(self.color_seed)
```